ebos/equil/initstateequil.cc

```diff
@@ -732,75 +741,106 @@ accountForScaledSaturations()
 {
     const auto gasActive = this->evalPt_.ptable->gasActive();
     const auto watActive = this->evalPt_.ptable->waterActive();
+    const auto oilActive = this->evalPt_.ptable->oilActive();
 
-    const auto& scaledDrainageInfo = this->matLawMgr_
-        .oilWaterScaledEpsInfoDrainage(this->evalPt_.position->cell);
-
-    const auto sg = this->sat_.gas;
-    const auto sw = this->sat_.water;
-
-    {
-        auto so = 1.0;
+    auto sg = gasActive? this->sat_.gas : 0.0;
+    auto sw = watActive? this->sat_.water : 0.0;
+    auto so = oilActive? this->sat_.oil : 0.0;
 
-        if (watActive) {
-            const auto swu = scaledDrainageInfo.Swu;
-            so -= swu;
-
-            this->fluidState_.setSaturation(this->waterPos(), swu);
-        }
-
-        if (gasActive) {
-            const auto sgu = scaledDrainageInfo.Sgu;
-            so -= sgu;
-
-            this->fluidState_.setSaturation(this->gasPos(), sgu);
-        }
+    this->fluidState_.setSaturation(this->waterPos(), sw);
+    this->fluidState_.setSaturation(this->oilPos(), so);
+    this->fluidState_.setSaturation(this->gasPos(), sg);
 
-        this->fluidState_.setSaturation(this->oilPos(), so);
-    }
+    const auto& scaledDrainageInfo = this->matLawMgr_
+        .oilWaterScaledEpsInfoDrainage(this->evalPt_.position->cell);
 
     const auto thresholdSat = 1.0e-6;
     if (watActive && ((sw + thresholdSat) > scaledDrainageInfo.Swu)) {
         // Water saturation exceeds maximum possible value.  Reset oil phase
         // pressure to that which corresponds to maximum possible water
         // saturation value.
         this->fluidState_.setSaturation(this->waterPos(), scaledDrainageInfo.Swu);
+        if (oilActive) {
+            this->fluidState_.setSaturation(this->oilPos(), so - sw + scaledDrainageInfo.Swu);
```

Comment:
How does this correctly adjust the oil saturation?  Suppose, for example, that inverting the capillary pressure curves gives
```
(Sw, So, Sg) = (0.4, 0.5, 0.1)
```
but that `Swu = 0.3`.  If we then reduce `Sw` to `Swu`, then I would **typically** expect the surplus 0.1 to be distributed among the other two phases, and if we assign the surplus entirely to the oil, then we **should** have
```
(Sw, So, Sg) = (0.3, 0.6, 0.1)
```
On the other hand, this expression gives
```
So = 0.5 - 0.4 + 0.3 = 0.4
```
from which we get the saturation distribution
```
(Sw, So, Sg) = (0.3, 0.4, 0.1)
```
which does not sum to one instead.  What am I missing?

Reply:
> What am I missing?

Just that I forgot to flip signs. I am surprised that this bug in the PR did not affect any of the tests.

Reply:
> I am surprised that this bug in the PR did not affect any of the tests.

I am too.  On the other hand, It's **possible** that the tests don't need this kind of adjustment at all, and that if they actually do then they mostly care about the water saturation and implicitly assume `So = 1 - Sw`.